nexus/src/app/sagas/mod.rs

```diff
@@ -304,3 +304,51 @@ pub(crate) use __action_name;
 pub(crate) use __emit_action;
 pub(crate) use __stringify_ident;
 pub(crate) use declare_saga_actions;
+
+/// Retry a progenitor client operation until a known result is returned.
+///
+/// Saga execution relies on the outcome of an external call being known: since
+/// they are idempotent, reissue the external call until a known result comes
+/// back. Retry if a communication error is seen.
+#[macro_export]
+macro_rules! retry_until_known_result {
+    ( $log:ident, $func:block ) => {{
+        use omicron_common::backoff;
+
+        backoff::retry_notify(
+            backoff::retry_policy_internal_service(),
+            || async {
+                match ($func).await {
+                    Err(progenitor_client::Error::CommunicationError(e)) => {
```

Comment:
I think there are other cases where we'd want to retry, like 503.  Take a look at this function:
https://github.com/oxidecomputer/omicron/blob/main/dns-service-client/src/lib.rs#L28

It feels like maybe want a generic progenitor-client `is_retryable()` function and we want to use that here.

I think this is still an improvement as-is.  We could defer this but I think we might want a TODO comment here or an issue suggesting that we unify this function and that other `is_retryable()` since they're separately implementing a very critical, non-trivial policy.

Reply:
Good point about the other cases, I've added a try on 503 and 429 [here](472f77a99a09db761ba75b18b5b1ab88f958938b).

I need to think a bit more about the other parts.